drafts/bitcartel-nu0-tx-format/draft.rst

```diff
@@ -0,0 +1,253 @@
+
+That is, it prevents a client on one branch of the network from attempting to parse transactions intended for another branch, in the situation where the transactions share the same format version number but are actually specified differently.  For example, Zcash and Zclone both define their own custom v3 transaction formats, but each will have its own unique version group id, so that they can reject v3 transactions with unknown version group ids.
+
+The combination of transaction version and version grouph id, `nVersion || nVersionGroupId` uniquely defines the transaction format, thus enabling parsers to reject transactions from outside the client's chain which cannot be parsed.  This helps provide users with a layer of replay protection at the parser level.  Full replay protection is defined in the Overwinter Transaction Signature Verification scheme [#zip-0???]_.
+
+It is expected that when introducing a new transaction version which requires a hard fork, a new unique version group id will be assigned to that transaction version.
+
+In the case that new transaction versions are soft-fork compatible with older transaction versions, the same version group id could be re-used.
+
+Expiry Height
+-------------
+
+The expiry height field specifies the last block height at which a transaction must be mined into a block, after which the transaction is deemed to have expired and should be removed from the mempool.  A block is invalid if it contains an expired transaction.  More information can be found in the Transaction Expiry ZIP [#zip-???]_
+
+Transaction Validation
+======================
+
+A valid Overwinter transaction intended for Zcash must have:
+
+- version number 3
+- overwinter flag set
+- version group id as specified in Zcash source code
+
+Overwinter transaction parsers should reject transactions from further processing and validation if the:
+
+- version number is unknown
+- overwinter flag is not set
+- version group id is not recognized
+
+Implementation
+==============
+
+Transaction Version
+-------------------
+
+Transaction version remains a positive value.
+
+Code can continue to check the transaction version using comparison operators::
+
+    if (tx.nVersion >= 2) {
+        for (int js = 0; js < joinsplits; js++) {
+
+Tests can continue to set the version to zero as an error condition::
+
+    mtx.nVersion = 0
+    
+Overwinter Validation
+---------------------
+
+To test if the format of an Overwinter transaction is valid or not::
+
+    if (tx.nVersion == 3 && tx.fOverwintered ) {
+        // Valid v3 format transaction
+    }
+
+To test if the format of an Overwinter transaction is intended for the client's chain::
+
+    if (tx.nVersion == 3 &&
+        tx.fOverwintered &&
+        tx.nVersionGroupID == OVERWINTER_VERSION_GROUP_ID) {
+        // Valid v3 format transaction intended for this client's chain
```

Comment:
This is the definition of a valid Overwinter transaction. We should check the transaction properties in order of precedence:

- `fOverwintered` switches between entire classes of version numbers.
- `nVersionGroupId` switches between groups of version numbers within a class.
- `nVersion` specifies a version number within a group.

```cpp
    if (tx.fOverwintered &&
        tx.nVersionGroupID == OVERWINTER_VERSION_GROUP_ID &&
        tx.nVersion == 3) {
        // Valid v3 format transaction intended for this client's chain
```